ecommerce/wagtail_views.py

```diff
@@ -0,0 +1,120 @@
+"""Wagtail admin views"""
+from wagtail.admin.views.generic.models import IndexView
+from wagtail.admin.viewsets.base import ViewSetGroup
+from wagtail.admin.viewsets.model import ModelViewSet
+from wagtail.permissions import ModelPermissionPolicy
+
+from ecommerce.models import Product, ProductVersion
+
+
+class CreateOnlyModelPermissionPolicy(ModelPermissionPolicy):
+    """
+    ModelPermissionPolicy to allow `add` permission.
+
+    Wagtail does not support a separate permission to allow read only mode.
+    The reason behind trying to go with a read-only mode is to avoid confusion
+    and allow only a single place to change the price of Products.
+    """
+
+    def user_has_permission(self, user, action):
+        """
+        Allows `add` permission
+        """
+        if action in ("change", "delete"):
+            return False
+        return super().user_has_permission(user, action)
+
+
+class ReadOnlyIndexView(IndexView):
+    """
+    IndexView to override the edit URL with read only inspect URL.
+    """
+
+    def get_edit_url(self, instance):
+        """
+        Returns inspect URL as we have disabled the edit permissions for the Product and ProductVersion.
+        """
+        return self.get_inspect_url(instance)
+
+
+class ProductViewSet(ModelViewSet):
+    """Wagtail ModelViewSet for Product"""
+
+    index_view_class = ReadOnlyIndexView
+    model = Product
+    search_fields = (
+        "courseruns__title",
+        "programs__title",
+        "courseruns__courseware_id",
+        "programs__readable_id",
+    )
+    form_fields = (
+        "id",
+        "content_type",
+        "object_id",
+        "is_active",
+        "is_private",
+    )
+    list_display = ("id", "content_object", "is_active", "price")
+    list_filter = (
+        "content_type",
+        "is_active",
+    )
+    inspect_view_enabled = True
+    icon = "pilcrow"
```

Comment:
We may or may not remove it. In Wagtail the nested lists doesn't have an icon unless they have child pages in which case the show a folder icon.

Reply:
I believe that I added this intentionally. Looked nice with the icons :)